samples/matter/basic/src/app_event.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <cstdint>
+
+struct AppEvent {
+
+	enum FunctionEventType : uint8_t { FunctionPress, FunctionRelease, FunctionTimer };
+
+	enum OtherEventType : uint8_t { StartThread = FunctionTimer + 1, StartBleAdvertising };
```

Comment:
`StartThread` won't be needed in this sample. You can actually merge these enum classes as all the events need no arguments.